### DISCUSSION TITLE
Support for multiple groups(arrays) of servos

### DISCUSSION THREAD

**Discussion by sjgcornish** [ArminJo/ServoEasing#71] [Q&A] [answered]
Support is provided for all servos with commands such as
synchronizeAllServosAndStartInterrupt
However there are situations where you want to have for example to have  pairs of continuous rotation servos (left and right drive) and two pairs of a pairs servos  (left and right steering, with servos in each pair synchronized. It would also be useful for a servo to belong to more than one group so for example the four drive servos could also be part of a group as well there left/right side group

Is there a way to support the above or does it require is additional enhancement.

**Comment by ArminJo** [comment, marked as answer]
This is not supported. you have to write your own functions like e.g. in https://github.com/ArminJo/ServoEasing/blob/058ddf029bfdb30e0af288f733b35f1bc1b35638/examples/QuadrupedControl/QuadrupedServoControl.hpp#L312

**Comment by sjgcornish** [comment]
Thanks for the quick response. I will follow the linked example.

**Comment by sjgcornish** [comment]
I apologise for troubling you with these questions.

I can see how these functions group the servos and the functions. Set all the servos wait until all set then move ALL on interrupt and then repeat the cycle.  My question is if the servo easeTo settings for a servo in the array have not changed since the last action is the action ignored.

synchronizeAllServosAndStartInterrupt(false)

The reason I ask is that some of the servos in the servo array are continuous servos so you do not want these to start easing again as they may have not reached the speed from the previous synchronizeAllServosAndStartInterrupt(false). For example in pan and tilt  or in my case 4 wheel drive motors. I guess if they are already at the desired (set speed) then a repeat eae to command would have no effect anyway.

**Reply by ArminJo** [reply]
Sice you know best which servo should move when and if they need to be synchronized, you have to rewrite the synchronizeAllServosAndStartInterrupt(false) by yourself or use another function like setEaseToForAllServos().

**Reply by sjgcornish** [reply]
Thanks for the pointer.  I can see that you are just iterating through the servo array, so as I know the array indexes of of the servos I can duplicate the ones I want to use with two extra parameters - groupStartIndex and groupEndIndex /or groupSize and add a few auxiliary functions to allow defining and managing groups
